### DISCUSSION TITLE
markdown environment inside beamer frame

### DISCUSSION THREAD

**Discussion by Ninlives** [Witiko/markdown#418] [Q&A] [answered]
The following snippet will not work:
```latex
\begin{frame}
\begin{markdown}
- A markdown list
\end{markdown}
\end{frame}
```
and end with the following error:
```
Runaway argument?
! Paragraph ended before ^^M was complete.
<to be read again>
                   \par
l.69 \end{frame}
?
! Emergency stop.
<to be read again>
                   \par
l.69 \end{frame}
```

Why is that? I'm unfamiliar with latex internals and I'm not sure what is going on here...

**Comment by Witiko** [comment, marked as answer]
@Ninlives Hi, thanks for using the Markdown package.

> The following snippet will not work:

You may need to use the `fragile` option on your frame:

``` tex
\begin{frame}[fragile]
\begin{markdown}
- A markdown list
\end{markdown}
\end{frame}
```

> Why is that?

Both `frame` and `markdown` are buffered environments, i.e. all text inside of them is read into a variable first and only then processed by TeX. However, whereas `frame` buffers the text as TeX would, i.e. `\command` is recognized as a command, `markdown` uses a different catcode regime, where most of the text is kept as-is, i.e. `\command` is recognized as regular text. Once a text has been buffered by `frame`, it cannot be correctly re-read by `markdown`.

Specifying the `fragile` option informs `frame` that it should tread carefully and store the text into an external file instead of trying to store it in a variable. This more roundabout way causes `markdown` and other buffered environments such as `verbatim` to work fine.

**Reply by Ninlives** [reply]
Thanks, this saves my problem, though I don't quite understand the reason 😕

**Reply by Witiko** [reply]
Glad to have helped. Sorry that I couldn't explain it better, perhaps [the beamer documentation][1] can do a better job:

![image](https://github.com/Witiko/markdown/assets/603082/e3b0e665-53c9-402d-9f18-a1db02234081)

 [1]: https://ctan.org/pkg/beamer
